src/routes/pay/stripe.ts

```diff
@@ -179,39 +180,57 @@ stripeRouter.post('/', async (req, res) => {
     )
   } catch (err) {
     logger.error(err)
+    slack.sendStripeAlert({
+      data: event,
+      message: 'Signature verification failed',
+    })
     res.status(400).send(`Webhook Error: ${err.message}`)
   }
 
   if (!event) {
+    slack.sendStripeAlert({ message: 'Empty event object' })
     return res.status(400).send(`Webhook Error: empty event object`)
   }
 
-  // Handle the event
-  switch (event.type) {
-    case 'payment_intent.canceled':
-      const canceled = event.data.object as Stripe.PaymentIntent
-      await updateTxState(canceled, event.type, canceled.cancellation_reason)
-      break
-    case 'payment_intent.payment_failed':
-      const failed = event.data.object as Stripe.PaymentIntent
-      await updateTxState(failed, event.type, failed.last_payment_error?.code)
-      break
-    case 'payment_intent.processing':
-    case 'payment_intent.succeeded':
-      const paymentIntent = event.data.object as Stripe.PaymentIntent
-      await updateTxState(paymentIntent, event.type)
-      break
-    case 'charge.refunded':
-      const charge = event.data.object as Stripe.Charge
-      await createRefundTxs(charge.refunds)
-      break
-    case 'customer.deleted':
-      const customer = event.data.object as Stripe.Customer
-      await paymentService.deleteCustomer({ customerId: customer.id })
-      break
-    default:
-      logger.error('Unexpected event type', event.type)
-      break
+  try {
+    // Handle the event
+    switch (event.type) {
+      case 'payment_intent.canceled':
+        const canceled = event.data.object as Stripe.PaymentIntent
+        await updateTxState(canceled, event.type, canceled.cancellation_reason)
+        break
+      case 'payment_intent.payment_failed':
+        const failed = event.data.object as Stripe.PaymentIntent
+        await updateTxState(failed, event.type, failed.last_payment_error?.code)
+        break
+      case 'payment_intent.processing':
+      case 'payment_intent.succeeded':
+        const paymentIntent = event.data.object as Stripe.PaymentIntent
+        await updateTxState(paymentIntent, event.type)
+        break
+      case 'charge.refunded':
+        const charge = event.data.object as Stripe.Charge
+        await createRefundTxs(charge.refunds)
+        break
+      case 'customer.deleted':
+        const customer = event.data.object as Stripe.Customer
+        await paymentService.deleteCustomer({ customerId: customer.id })
+        break
+      default:
+        logger.error('Unexpected event type', event.type)
+        slack.sendStripeAlert({
+          data: event,
+          message: 'Unexpected event type',
+        })
+        break
+    }
+  } catch (error) {
+    logger.error(error)
+    slack.sendStripeAlert({
+      data: event,
```

Comment:
Should  we pass event data like [queue does](https://github.com/thematters/matters-server/pull/1734/files#diff-4976b7a1365825fb091212aef498537e539e36c5da1b9ed24b21f6a29d805788R60-R64) since the event may contain sensitive data?

Reply:
Revised. Thanks.